pyresample/geometry.py

```diff
@@ -2342,12 +2342,22 @@ def get_lonlats(self, nprocs=None, data_slice=None, cache=False, dtype=None, chu
             row_slice = slice(0, self.height)
             col_slice = slice(0, self.width)
         offset = 0
-        for definition in self.defs:
+        for def_idx, definition in enumerate(self.defs):
+            # handle case of stacked area definitions where chunks need to be assigned to each areadef
+            if isinstance(chunks, tuple) and isinstance(chunks[0], int):
+                chunks_for_definition = chunks
+            elif isinstance(chunks, tuple) and len(chunks[0]) != len(self.defs):
+                chunks_for_definition = (definition.shape[0], chunks[1])
+            elif isinstance(chunks, tuple) and len(chunks[0]) == len(self.defs):
+                chunks_for_definition = (chunks[0][def_idx], chunks[1])
+            else:
+                chunks_for_definition = chunks
```

Comment:
For clarity, could you put this in a separate method?

Reply:
Done!